pkg/kubelet/stats/pidlimit/pidlimit_linux.go

```diff
@@ -39,12 +41,33 @@ func Stats() (*statsapi.RlimitStats, error) {
 		}
 	}
 
-	var info syscall.Sysinfo_t
-	syscall.Sysinfo(&info)
-	procs := int64(info.Procs)
-	rlimit.NumOfRunningProcesses = &procs
+	if procs, err := runningProcNum(); err == nil {
+		rlimit.NumOfRunningProcesses = &procs
+	} else {
+		var info syscall.Sysinfo_t
+		syscall.Sysinfo(&info)
+		procs := int64(info.Procs)
+		rlimit.NumOfRunningProcesses = &procs
+	}
```

Comment:
If opening `/proc/loadavg` fails, won't this bear a risk of giving an erroneous reading?

Reply:
Yes, however it doesn't make thing worse compared to previous code. 

Another possibility is that /proc/loadavg might be open and read successfully, but the content has been changed in format so the functions returns an error. 

I might be too conservative to keep the sysinfo code here.

Reply:
I think it makes sense to keep it... at least for compatibility with older kernels?
I guess it's not easy to add a test with 65k+ processes to show the benefits of this patch?
/ok-to-test

Reply:
Maybe we can start a pod that creates 65k+ tasks (on Linux, POSIX threads implemented as kernel tasks consume a PID each)? It's definitely possible to make that container image.

Reply:
I made a tool that might help with this testing: https://github.com/sftim/manytasks

Reply:
rust XD
@sftim looks good, however not sure if that fits our e2e testing... at least for local tinkering for sure it helps

Reply:
> rust

The prototype was Perl...